3p-ecosystem/src/main/java/com/google/homesampleapp/screens/inspect/InspectScreen.kt

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.homesampleapp.screens.inspect
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.LifecycleResumeEffect
+import androidx.navigation.NavController
+import com.google.homesampleapp.Device
+import com.google.homesampleapp.R
+import com.google.homesampleapp.chip.DeviceMatterInfo
+import com.google.homesampleapp.chip.MatterConstants
+import com.google.homesampleapp.screens.common.DialogInfo
+import com.google.homesampleapp.screens.common.MsgAlertDialog
+import com.google.protobuf.Timestamp
+import timber.log.Timber
+
+/**
+ * The Inspect Screen shows all the "cluster" information about the currently selected device in the
+ * Device screen.
+ */
+@Composable
+fun InspectRoute(
+  navController: NavController,
+  innerPadding: PaddingValues,
+  deviceId: Long,
+  inspectViewModel: InspectViewModel = hiltViewModel(),
+) {
+  Timber.d("InspectRoute deviceId [$deviceId]")
+
+  // Controls the Msg AlertDialog.
+  // When the user dismisses the Msg AlertDialog, we "consume" the dialog.
+  val msgDialogInfo by inspectViewModel.msgDialogInfo.collectAsState()
+  val onDismissMsgDialog: () -> Unit = { inspectViewModel.dismissMsgDialog() }
```

Comment:
This needs a remember, since the `ViewModel` instance read is unstable from a compose compiler POV:

```
val onDismissMsgDialog: () -> Unit = remember {
    { inspectViewModel.dismissMsgDialog() }
}
```

Reply:
OK, I asked bard, and my understanding is now the following:
It looks like the only scenarios where I should generally avoid using remember for values passed to other Composables in Jetpack Compose:
1. Simple, Immutable Values
Primitive Types: For basic data types like Int, String, Boolean, there's no need for remember. These are inherently immutable, so their instances never change.
Data Classes: If your data class contains only primitive types or other immutable values, it's safe to pass it directly without remember.
2. Values Derived Without Expensive Computations
3. Frequent Changes Intended to Trigger Recomposition
4. Values Managed by State Holders

Now regarding lambdas. Let's say I have this in a parent composable:
    val onClick: () -> Unit = { println("hello") }
and it is passed as an argument to a child composable.
If the parent composable is recomposed, would that onClick be considered unstable? If so, why?

Reply:
Read more about remember, I think I have a better understanding now.
My latest commit does wrap with "remember" all lambdas that are passed as arguments to other composables.

Reply:
Lambdas in compose are stable types, regardless of what objects are captured in its closure. However, Since the Composable where the lambda is declared can be recomposed, you will end up creating multiple instances of the lambda which may (depending on how often the containing composable is recomposed) cause unnecessary GC churn. 

So the remember here is just to make sure the lambda is allocated once and only once. Since the lambda effectively captures the VM instance, this is safe.

Reply:
Got it. Thx for the clarifications, makes sense now.